src/components/Integrations/IntegrationDetails.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import {
+  DescriptionList,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+} from '@patternfly/react-core';
+import {
+  IntegrationCamel,
+  IntegrationIcon,
+  IntegrationPagerduty,
+  IntegrationType,
+} from '../../types/Integration';
+import { IntegrationStatus, StatusUnknown } from './Table/IntegrationStatus';
+import { IntegrationRow } from './Table';
+import Config, { defaultIconList } from '../../config/Config';
+import messages from '../../properties/DefinedMessages';
+import { useIntl } from 'react-intl';
+
+export const getIntegrationIcon = (type: string): React.ReactElement | null => {
+  const allIcons = Object.assign(
+    defaultIconList['Communications'],
+    defaultIconList['Reporting']
+  );
+  const integrationType: IntegrationIcon | undefined = Object.values(
+    allIcons
+  ).find((icon: IntegrationIcon) => icon.name === type);
+  if (integrationType === undefined) {
+    return null;
+  }
+  return (
+    <>
+      <img
+        width="16px"
+        src={integrationType.icon_url}
+        alt={integrationType.product_name}
+        className="pf-v5-u-mr-sm"
+      />
+    </>
+  );
+};
+
+interface IntegrationDetailsProps {
+  integration?: IntegrationRow;
+}
+
+const IntegrationDetails: React.FunctionComponent<IntegrationDetailsProps> = ({
+  integration,
+}) => {
+  const intl = useIntl();
+
+  const buildIntegrationDetails = (integration: IntegrationRow) => {
```

Comment:
Can we break this into separate components? That way when new field needs to be added we can just adjust the component.